### ISSUE TITLE
[BUG] ANNOTATION location rules not working

### ISSUE THREAD

**Issue by abrugaro** [konveyor/analyzer-lsp#901]
We have 2 `Producer annotation no longer required rules.` 

One checks for `javax.enterprise.inject.Produces`: https://github.com/konveyor/rulesets/blob/5598b6eb695d496d9f39c98dfbcab32c25623faf/default/generated/quarkus/210-cdi-to-quarkus.windup.yaml#L65

And the other for `jakarta.enterprise.inject.Produces`: https://github.com/konveyor/rulesets/blob/5598b6eb695d496d9f39c98dfbcab32c25623faf/default/generated/quarkus/212-jakarta-cdi-to-quarkus.windup.yaml#L65

Both rules are raising incidents in https://github.com/konveyor-ecosystem/coolstore/blob/main/src/main/java/com/redhat/coolstore/rest/CartEndpoint.java#L49 but that annotation is `import javax.ws.rs.Produces`

In this case, the analyzer was correctly identifying that there were an annotation but the fqdn didn't match.

---

In this other case, the repository https://github.com/ibraginsky/book-server
 is analyzed with the following command:

```
kantra analyze --overwrite --input /home/abrugaro/PyCharmProjects/kantra-cli-tests/data/applications/book-server --output /home/abrugaro/Desktop/kantra-annotation-report --source java-ee --source springboot --target cloud-readiness --target linux --target quarkus
```

The rule with ID `springboot-jmx-to-quarkus-00001` identifies 9 incidents.

That rule (https://github.com/konveyor/rulesets/blob/5598b6eb695d496d9f39c98dfbcab32c25623faf/default/generated/quarkus/228-springboot-jmx-to-quarkus.windup.yaml#L33
) is defined as follows:

```yaml
when:
    java.referenced:
      location: ANNOTATION
      pattern: org.springframework.jmx.*
```

However, it is being triggered in places that don’t even contain annotations, such as:
https://github.com/ibraginsky/book-server/blob/main/src/main/java/com/telran/application/model/dto/Book.java#L17

<img width="1901" height="863" alt="Image" src="https://github.com/user-attachments/assets/589b7d6a-593f-42c9-948e-188483db36c3" />

**Comment by jmle** [comment]
@shawn-hurley this should be fixed by now I think

**Comment by shawn-hurley** [comment]
Closing this as fixed by the updates to java-analyzer-bundle and is tested in konveyor/java-analyzer-bundle#165
